### PR TITLE
[submodule] update protobuf to tag v3.19.6

### PR DESCRIPTION
**Pull request by cyyever**
Fixes #ISSUE_NUMBER

### CODE REVIEW

**Comment by Skylion007** [comment]
I was going to get around to this at some point. You will also want to update the pip install pinned versions of protobuf to the latest as well. (do a grep through the yaml and requirement files).

**Comment by vadimkantorov** [comment]
There is also this issue requiring protobuf < 3.2.0 for tensorboard to work: https://github.com/pytorch/pytorch/issues/90374

**Comment by cyyever** [comment]
> I was going to get around to this at some point. You will also want to update the pip install pinned versions of protobuf to the 21.12 as well. (do a grep through the yaml and requirement files).

No pip dependency found after a grep. Bazel files were also checked but no specific versions hard-wired.

**Comment by cyyever** [comment]
> There is also this issue requiring protobuf < 3.2.0 for tensorboard to work: #90374. I propose there that torch.utils.tensorboard should not fail on import because of problematic protobuf dependencies

May be we need to upgrade tensorboard dependency as well.

File: requirements.txt [hunk 1]
```diff
@@ -13,3 +13,4 @@ sympy
 filelock
 networkx
 jinja2
+protobuf==4.21.12
```
Comment (Skylion007) [inline requirements.txt:16]: Should be a min requirement though, right?

**Comment by vadimkantorov** [comment]
> May be we need to upgrade tensorboard dependency as well.

As far as I know, tensorboard still has this problem unfixed. So you could bump the min version of tensorboard, but it still wouldn't fix protobuf <3.2.0 requirement of tensorboard: https://github.com/tensorflow/tensorboard/pull/6147 v3.19.6 seems to be the most recent supported version

**Review by malfet — CHANGES_REQUESTED** [review]
For those kinds of changes, I want some rigorous and reproducible test plan, like a script that dumps public symbols on all 3 platforms(with priority on Linux and Windows) and showing that no new ones were added.

Another good test to add (and may be to CI): Install torch + different version of protobuf from pipy and make sure they do not conflict with each other, as all the ugly patching that you are removing was done in order to make protobuf not leak any symbols externally.

**Pull request closed without merging**